src/PrestaShopBundle/Twig/Extension/SymfonyLayoutExtension.php

```diff
@@ -0,0 +1,146 @@
+use PrestaShopBundle\Bridge\SymfonyLayoutFeature;
+use PrestaShopBundle\EventListener\ContextShopListener;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Twig\Extension\AbstractExtension;
+use Twig\Extension\GlobalsInterface;
+
+/**
+ * Handles twig variables for pure symfony layout (that replaces the old legacy layout),
+ * it is tightly coupled with ContextShopListener which is responsible for creating the ControllerConfiguration
+ * and set the ControllerBridge on Context->controller, the controller configuration is then accessible via a request
+ * attribute.
+ *
+ * These components/listeners were initially created for horizontal migration (which has been
+ * abandoned since) but it turns out they fit the need we have for Symfony layout especially
+ * handling most of the variables previously initialized by AdminLegacyLayoutController.
+ */
+class SymfonyLayoutExtension extends AbstractExtension implements GlobalsInterface
+{
+    /** @var LegacyContext */
+    private $context;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var ConfiguratorInterface[]
+     */
+    private $configurators;
+
+    /**
+     * @var SymfonyLayoutFeature
+     */
+    private $symfonyLayoutFeature;
+
+    public function __construct(
+        LegacyContext $context,
+        RequestStack $requestStack,
+        iterable $configurators,
+        SymfonyLayoutFeature $symfonyLayoutFeature
+    ) {
+        $this->context = $context;
+        $this->requestStack = $requestStack;
+        $this->configurators = $configurators;
+        $this->symfonyLayoutFeature = $symfonyLayoutFeature;
+    }
+
+    public function getGlobals(): array
+    {
+        $useSymfonyLayout = $this->symfonyLayoutFeature->isEnabled();
+        $layoutVariables = [];
+
+        if ($this->symfonyLayoutFeature->isEnabled()) {
+            $request = $this->requestStack->getCurrentRequest();
+            $controllerConfiguration = $request->attributes->get(ContextShopListener::CONTROLLER_CONFIGURATION_ATTRIBUTE);
+            if ($controllerConfiguration instanceof ControllerConfiguration) {
+                foreach ($this->configurators as $configurator) {
+                    $configurator->configure($controllerConfiguration);
+                }
+                $layoutVariables = $controllerConfiguration->templateVars + $this->renderSmartyContent($controllerConfiguration);
+            }
+        }
+
+        return $layoutVariables + [
+            'use_legacy_layout' => !$useSymfonyLayout,
+            'use_symfony_layout' => $useSymfonyLayout,
+        ];
+    }
+
+    private function renderSmartyContent(ControllerConfiguration $controllerConfiguration): array
+    {
+        $smarty = $this->context->getSmarty();
+        $smarty->assign($controllerConfiguration->templateVars);
+        $smarty->setTemplateDir([
+            _PS_BO_ALL_THEMES_DIR_ . 'new-theme' . DIRECTORY_SEPARATOR . 'template',
```

Comment:
@FabienPapet I tried to add this correctly but it has too many side effects, create-test-db install cli (an so all UI tests) fail because the const is not correctly defined in those context Mostly because it depends on ADMIN_DIR

So I think it should be handled in a dedicated PR